vue-components/src/BoxAnnotations.vue

```diff
@@ -45,35 +62,50 @@ const lineWidthInDisplay = computed(
 );
 
 // Draw visible annotations
-watchEffect(() => {
-  if (!visibleCanvas.value || !visibleCtx.value || !rect.value) return;
+watch(
+  [
+    visibleCanvas,
+    visibleCtx,
+    canvasDims,
+    lineWidthInDisplay,
+    lineOpacity,
+    imageToCanvasScale,
+    () => props.boxAnnotations,
+  ],
```

Comment:
If `drawVisibleAnnotations` starts using a new compnent wide ref and we forget to add it here, there may be bug.  But it is nice you facotred out the function.  Maybe we should just pass all the resolved refs as args to `drawVisible...`